### PR TITLE
chore(controller): controller chart main storage supports local fs

### PR DESCRIPTION
**Pull request by anda-ren**
## Description

- controller fs chart support;
- fix put error when device not the same
- fix path contains slash

chart command:
```
helm upgrade --install controller-fs . -n controller-fs --create-namespace \
  --set image.registry=10.131.0.1:8083 \
  --set image.tag=resume_status-e19d1e21 \
  --set mirror.pypi.enabled=true \
  --set mirror.pypi.indexUrl=http://10.131.0.1:8081/repository/pypi-grouped/simple \
  --set mirror.pypi.trustedHost=10.131.0.1 \
  --set mysql.image=docker-registry.starwhale.cn/bitnami/mysql:8.0.29-debian-10-r2 \
  --set mysql.initImage=docker-registry.starwhale.cn/bitnami/bitnami-shell:11-debian-11-r6 \
  --set minio.enabled=false \
  --set devMode.createPV.enabled=true \
  --set devMode.createPV.host=host005-bj01 \
  --set resources.controller.limits.memory=16G \
  --set devMode.createPV.rootPath=/mnt/data/starwhale/controller-fs/controller-fs \
  --set controller.ingress.host=controller-fs.pre.intra.starwhale.ai \
  --set controller.storageType=fs
 ```

## verification
http://controller-fs.pre.intra.starwhale.ai/projects/1/evaluations/6/results
## Modules
- [ ] UI
- [ ] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [ ] run code format and lint check
- [ ] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: docker/charts/values.yaml [hunk 1]
```diff
@@ -66,14 +66,18 @@ externalOSS:
...
   region: "local"
-  type: "s3" # supported type: s3, minio and aliyun
 
 controller:
   taskSplitSize: 2
   jwt:
     tokenExpireMinutes: 43200
   upload:
     maxFileSize: 20480MB
+  storageType: "s3" # supported type: s3/minio/aliyun/fs
```
Comment (tianweidut) [inline docker/charts/values.yaml:76]:
```
  storageType: "s3" # supported type: s3/minio/aliyun/fs
  fs:
    rootdir: /usr/local/starwhale
  persistence:
    size: 50Gi
```
-->
```
  storageType: "minio"   # supported type: s3/minio/aliyun/fs
  fsStoragePV:
     size: 50Gi
     rootPath: "/var/starwhale/local-storage/fs"
```
?

**Comment by tianweidut** [comment]
@anda-ren  update the controller.storageType field in the [values.minikube.cn.yaml](https://github.com/star-whale/starwhale/blob/main/docker/charts/values.minikube.cn.yaml) and [values.minikube.global.yaml)](https://github.com/star-whale/starwhale/blob/main/docker/charts/values.minikube.global.yaml) ?

File: docker/charts/values.yaml (hunk 1 above)
Comment (anda-ren) [reply docker/charts/values.yaml]: done

**Comment by anda-ren** [comment]
> 

done

File: docker/charts/templates/pv.yaml [hunk 2]
```diff
@@ -1,8 +1,12 @@
 {{ if .Values.minikube.enabled}}
 
+{{ if eq .Values.controller.storageType "fs"}}
+{{ include "chart.minikubePV" (merge (dict "backend" "controller") . )}}
+{{else}}
```
Comment (jialeicui) [inline docker/charts/templates/pv.yaml:5]: Add spaces?

File: docker/charts/README.md [hunk 3]
```diff
@@ -88,7 +88,7 @@ The `update` command will update the information of available charts locally fro
 | `controller.ingress.ingressClassName` | ingress class name | `nginx`|
 | `controller.ingress.host` | Starwhale controller domain | `console.pre.intra.starwhale.ai` |
 | `controller.containerPort`| Starwhale console web port | `8082` |
-| `controller.storageType` | Starwhale supports `s3`, `minio`, `aliyun` and `fs` as the main storage. | `s3` |
+| `controller.storageType` | Starwhale supports `s3`, `minio`, `aliyun` and `fs` as the main file storage. | `s3` |
```
Comment (jialeicui) [inline docker/charts/README.md:91]: Change the default to `minio`?

Comment (anda-ren) [reply docker/charts/README.md]: done

File: docker/charts/templates/pv.yaml (hunk 2 above)
Comment (anda-ren) [reply docker/charts/templates/pv.yaml]: done

File: docker/charts/templates/controller-deployment.yaml [hunk 4]
```diff
@@ -67,7 +75,7 @@ spec:
               value: controller
               {{- if not .Values.minikube.enabled }}
             - name: JVM_XMX
-              value: {{ .Values.resources.controller.requests.memory }}
+              value: {{ .Values.resources.controller.limits.memory }}
```
Comment (jialeicui) [inline docker/charts/templates/controller-deployment.yaml:78]: Maybe we could remove this block?
ref: https://www.eclipse.org/openj9/docs/xxusecontainersupport/

Comment (anda-ren) [reply docker/charts/templates/controller-deployment.yaml]: yes,done

**Review by tianweidut — COMMENTED** [review]
LGTM

**Review by jialeicui — APPROVED** [review]
LGTM

**Pull request merged**